libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -291,8 +291,11 @@ void ArrayUtils::copyArrayToMemory(ArrayType const& _sourceType, bool _padToWord
 	CompilerUtils utils(m_context);
 	unsigned baseSize = 1;
 	if (!_sourceType.isByteArray())
+	{
 		// We always pad the elements, regardless of _padToWordBoundaries.
 		baseSize = _sourceType.baseType()->calldataEncodedSize();
+		solAssert(baseSize >= 0x20, "");
```

Comment:
Can you explain why is this needed?

Reply:
This might want to copy weird things like arrays of mappings. If baseSize is zero, then converting length into size and then looping over the size might not terminate.